docs/simpledroidble/usage.rst

```diff
@@ -9,39 +9,103 @@ This code is currently under active development and some features are not yet
+    - ``simpleble/src/backends/android/simpleble-bridge/build/outputs/aar/simpleble-bridge-release.aar``
+
+3. Follow `Add SimpleDroidBLE to your project`_.
+
+Add SimpleDroidBLE to your project
+==================================
+
+From AAR file
+-------------
+
+1. Place the package on your app's directory, and take note of its path. We'll use ``example-app/app/libs/{simpledroidble,simpleble-bridge}.aar``.
+
+2. Declare the dependency on your app's ``build.gradle{.kts}`` file (note: relative paths begin from your app's module folder)::
+
+    dependencies {
+      /* ... */
+      implementation(files('libs/simpledroidble.aar', 'libs/simpleble-bridge.aar'))
+    }
+
+3. You're ready to start using SimpleBLE on Android! Go to `Using SimpleDroidBLE`_ to continue with the onboarding.
+
 Consuming Locally
-=================
+-----------------
 
 If you want to use SimpleDroidBLE as part of your project from a local copy,
-you can do so by adding the following to your `settings.gradle` or `settings.gradle.kts`file.
-Make sure this include is before your `include(":app")` statement.
+you can do so by adding the following to your ``settings.gradle`` or ``settings.gradle.kts`` file.
+Make sure this include is before your ``include(":app")`` statement::
 
-```groovy
-includeBuild("path/to/simpledroidble") {
-    dependencySubstitution {
-        substitute module("org.simpleble.android:simpledroidble") with project(":simpledroidble")
+    includeBuild("path/to/simpledroidble") {
+        dependencySubstitution {
+            substitute module("org.simpleble.android:simpledroidble") with project(":simpledroidble")
+        }
     }
-}
 
-```kotlin
-includeBuild("path/to/simpledroidble") {
-    dependencySubstitution {
-        substitute(module("org.simpleble.android:simpledroidble")).using(project(":simpledroidble"))
+Or, in Kotlin::
+
+    includeBuild("path/to/simpledroidble") {
+        dependencySubstitution {
+            substitute(module("org.simpleble.android:simpledroidble")).using(project(":simpledroidble"))
+        }
     }
-}
-```
-
-Then, inside your `build.gradle` or `build.gradle.kts` file, you can add the
-following dependency:
-
-```groovy
-dependencies {
-    implementation "org.simpleble.android:simpledroidble"
-}
-```
-
-```kotlin
-dependencies {
-    implementation("org.simpleble.android:simpledroidble")
-}
-```
\ No newline at end of file
+
+Then, inside your ``build.gradle`` or ``build.gradle.kts`` file, you can add the
+following dependency::
+
+    dependencies {
+        implementation "org.simpleble.android:simpledroidble"
+    }
+
+Kotlin::
+
+    dependencies {
+        implementation("org.simpleble.android:simpledroidble")
+    }
+
+Using SimpleDroidBLE
+====================
+
+First you'll need to provide ``SimpleDroidBLE.contextReference`` with a WeakReference to the app's context. In this step, the native code will be loaded in memory and permissions will be requested to the user.
```

Comment:
This needs to be a lot more elaborate. Why is this required? How do you do it? Show a few code examples.